domain/annotation/state/state_test.go

```diff
@@ -0,0 +1,404 @@
+// Keeping these explicit here should ensure we quickly detect any changes in the future
+func (s *stateSuite) TestKindNameFromID(c *gc.C) {
+
+	t1, err := kindNameFromID(annotations.ID{Kind: annotations.KindMachine, Name: "foo"})
+	c.Assert(err, jc.ErrorIsNil)
+	c.Check(t1, gc.Equals, "machine")
+
+	t2, err := kindNameFromID(annotations.ID{Kind: annotations.KindUnit, Name: "foo"})
+	c.Assert(err, jc.ErrorIsNil)
+	c.Check(t2, gc.Equals, "unit")
+
+	t3, err := kindNameFromID(annotations.ID{Kind: annotations.KindApplication, Name: "foo"})
+	c.Assert(err, jc.ErrorIsNil)
+	c.Check(t3, gc.Equals, "application")
+
+	t4, err := kindNameFromID(annotations.ID{Kind: annotations.KindStorage, Name: "foo"})
+	c.Assert(err, jc.ErrorIsNil)
+	c.Check(t4, gc.Equals, "storage_instance")
+
+	t5, err := kindNameFromID(annotations.ID{Kind: annotations.KindCharm, Name: "foo"})
+	c.Assert(err, jc.ErrorIsNil)
+	c.Check(t5, gc.Equals, "charm")
+
+	t6, err := kindNameFromID(annotations.ID{Kind: annotations.KindModel, Name: "foo"})
+	c.Assert(err, jc.ErrorIsNil)
+	c.Check(t6, gc.Equals, "model")
+
+	_, err = kindNameFromID(annotations.ID{Kind: 12, Name: "foo"})
+	c.Assert(err, jc.ErrorIs, annotationerrors.UnknownKind)
+
+}
+
+// ensureAnnotation is a test utility that manually adds a row to an annotation table
+//
+// s.manuallyInsertAnnotations("machine", "uuid123", "keyfoo", "valuebar") will add the row
+// (uuid123 keyfoo valuebar) into the annotation_machine table
+//
+// If the id is model, it'll just ignore the uuid and add the key value pair into the
+// annotation_model table
+func (s *stateSuite) ensureAnnotation(c *gc.C, id, uuid, key, value string) {
+	if id == "model" {
+		err := s.TxnRunner().StdTxn(context.Background(), func(ctx context.Context, tx *sql.Tx) error {
+			_, err := tx.ExecContext(ctx, `
+				INSERT INTO annotation_model (key, value)
+				VALUES (?, ?)`, key, value)
+			return err
+		})
+		c.Assert(err, jc.ErrorIsNil)
+	} else {
+		err := s.TxnRunner().StdTxn(context.Background(), func(ctx context.Context, tx *sql.Tx) error {
+			_, err := tx.ExecContext(ctx, fmt.Sprintf(`
+				INSERT INTO annotation_%[1]s (uuid, key, value)
+				VALUES (?, ?, ?)`, id), uuid, key, value)
+			return err
+		})
+		c.Assert(err, jc.ErrorIsNil)
+	}
+}
+
+// ensureMachine manually inserts a row into the machine table.
+func (s *stateSuite) ensureMachine(c *gc.C, id, uuid string) {
+	s.ensureNetNode(c, "node2")
+	s.ensureLife(c, "life3", "3")
+
+	err := s.TxnRunner().StdTxn(context.Background(), func(ctx context.Context, tx *sql.Tx) error {
+		_, err := tx.ExecContext(ctx, `
+		INSERT INTO machine (uuid, net_node_uuid, machine_id, life_id)
+		VALUES (?, "node2", ?, "life3")`, uuid, id)
+		return err
+	})
+	c.Assert(err, jc.ErrorIsNil)
+}
+
+// ensureApplication manually inserts a row into the application table.
+func (s *stateSuite) ensureApplication(c *gc.C, name, uuid string) {
+	s.ensureLife(c, "life3", "3")
```

Comment:
Don't do this. The schema will already have life IDs that you can use; 0=alive for example.